sycl/include/sycl/handler.hpp

```diff
@@ -381,6 +381,15 @@ class __SYCL_EXPORT handler {
                             "a single kernel or explicit memory operation.");
   }
 
+  enum SyclExtensions {
+    sycl_ext_oneapi_kernel_properties,
+    sycl_ext_oneapi_enqueue_barrier,
+    sycl_ext_oneapi_memcpy2d,
+    sycl_ext_oneapi_device_global
+  };
```

Comment:
Looks much better! This probably has to go somewhere into the `detail` or `ext::oneapi::experimental` namespace.

Reply:
This could use a comment as well, it's not immediately clear to the reader (outwith the context of this PR) why this subset of extensions is chosen.

Reply:
I moved this definition to ext::oneapi::experimental::detail (where is defined graph_impl) and added some comments.